tmc-langs-ant/src/main/java/fi/helsinki/cs/tmc/langs/ant/AntPlugin.java

```diff
@@ -21,23 +36,160 @@ public String getLanguageName() {
     }
 
     @Override
-    public ImmutableList<Path> findExercises(Path basePath) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    public ExerciseDesc scanExercise(Path path, String exerciseName) {
+        if (!isExerciseTypeCorrect(path)) {
+            return null;
+        }
+
+        List<String> output = startProcess(buildTestScannerArgs(path, null));
+        String outputString = "";
+
+        for (String line : output) {
+            outputString += line;
+        }
+        return resultParser.parseScannerOutput(outputString, exerciseName);
     }
 
     @Override
-    public ExerciseDesc scanExercise(Path path, String exerciseName) {
-        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
+    protected boolean isExerciseTypeCorrect(Path path) {
+        return new File(path.toString() + File.separatorChar + "build.xml").exists();
     }
 
     @Override
-    public RunResult runTests(Path path) {
-        ArrayList<String> args = new ArrayList<>();
-        throw new UnsupportedOperationException("Not supported yet.");
+    public ValidationResult checkCodeStyle(Path path) {
+        try {
+            CheckstyleRunner runner = new CheckstyleRunner(path.toFile(), new Locale("fi"));
+            return runner.run();
+        } catch (TMCCheckstyleException ex) {
+            log.log(Level.SEVERE, "Error running checkstyle:", ex);
+            return null;
+        }
     }
 
     @Override
-    public boolean isExerciseTypeCorrect(Path path) {
-        return new File(path.toString() + File.separatorChar + "build.xml").exists();
+    public RunResult runTests(Path path) {
+        buildAntProject(path);
+        List<String> runnerArgs = buildTestRunnerArgs(path);
+        startProcess(runnerArgs);
+
+        try {
+            FileUtils.deleteDirectory(new File(path.toString() + File.separatorChar + "build"));
+        } catch (IOException e) {
+            throw Throwables.propagate(e);
+        }
+
+        File resultFile = new File(path.toString() + resultsFile);
+        RunResult result = resultParser.parseTestResult(resultFile);
+        resultFile.delete();
+
+        return result;
+    }
+
+    /**
+     * Runs the build.xml file for the the given exercise.
+     *
+     * @param path The file path of the exercise directory.
+     */
+    public void buildAntProject(Path path) {
```

Comment:
Added @VisibleForTesting annotation and decreased the visibility to package-protected.